app/shared/ui-settings/src/commonMain/kotlin/ui/settings/tabs/theme/ThemePreferences.kt

```diff
@@ -0,0 +1,291 @@
+                contentPadding = PaddingValues(horizontal = 12.dp),
+            ) { page ->
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.Center,
+                ) {
+                    colors[page].forEach { color ->
+                        ColorButtons(
+                            color = color,
+                            themeSettings = themeSettings,
+                            state = state,
+                        )
+                    }
+                }
+            }
+
+            Row(
+                Modifier
+                    .fillMaxWidth()
+                    .align(Alignment.CenterHorizontally)
+                    .padding(vertical = 12.dp)
+                    .clearAndSetSemantics {},
+                horizontalArrangement = Arrangement.Center,
+            ) {
+                repeat(pageCount) { iteration ->
+                    val color = if (pagerState.currentPage == iteration)
+                        MaterialTheme.colorScheme.primary
+                    else
+                        MaterialTheme.colorScheme.outlineVariant
+                    Box(
+                        modifier = Modifier
+                            .padding(2.dp)
+                            .size(6.dp)
+                            .clip(CircleShape)
+                            .background(color),
+                    )
+                }
+            }
+        } else {
+            FlowRow(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.Center,
+                // maxItemsInEachRow = 4
+            ) {
+                colorList.forEach { color ->
+                    ColorButtons(
+                        color = color,
+                        themeSettings = themeSettings,
+                        state = state,
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun ColorButtons(
+    color: Color,
+    themeSettings: ThemeSettings,
+    state: SettingsState<ThemeSettings>
+) {
+    ColorButtonImpl(
+        modifier = Modifier,
+        isSelected = { color.toArgb() == themeSettings.seedColor && !themeSettings.useDynamicTheme },
+        cardColor = MaterialTheme.colorScheme.surfaceContainer,
+        containerColor = MaterialTheme.colorScheme.primaryContainer,
+        onClick = {
+            state.update(
+                themeSettings.copy(
+                    seedColor = color.toArgb(),
+                    useDynamicTheme = false,
+                ),
+            )
+        },
+        baseColor = color,
+    )
+}
+
+@Composable
+fun ColorButtonImpl(
+    modifier: Modifier = Modifier,
+    isSelected: () -> Boolean = { false },
+    cardColor: Color = MaterialTheme.colorScheme.primaryContainer,
+    containerColor: Color = MaterialTheme.colorScheme.primaryContainer,
+    onClick: () -> Unit = {},
+    baseColor: Color
+) {
```

Comment:
- `onClick` 作为一个必须动作 (无论用户传递什么值, UI 都会是可以点击的), 不应当有默认值. `isSelected` 同理
- boolean 参数名字如果能用过去式就不带 `is`
- composable 参数顺序是: callback < 必要参数 < modifier < 可选参数 < content lambda. 越小的在越前
- 这个看起来是一个完整组件, 不需要叫做 Impl. 可以把上面的改成重载.

```suggestion
@Composable
fun ColorButton(
    onClick: () -> Unit,
    baseColor: Color,
    selected: () -> Boolean,
    modifier: Modifier = Modifier,
    cardColor: Color = MaterialTheme.colorScheme.primaryContainer,
    containerColor: Color = MaterialTheme.colorScheme.primaryContainer,
) {
```